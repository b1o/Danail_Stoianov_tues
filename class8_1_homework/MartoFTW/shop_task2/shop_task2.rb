require "csv"
require "date"

our_date = Date.strptime(ARGV[1].to_s,'%d/%m/%Y')

arr_out = []

	CSV.foreach(ARGV[0]) do |row|

		if Date.strptime(row[3].to_s,'%d/%m/%Y') <= our_date

			arr_out << row		

		end

	end

	arr_out = arr_out.sort_by{|a,b,c,d| b}

	CSV.open("shop_task2_result.csv","w") do |csv|
		arr_out.each do |elem|
			csv << elem 
		end
	end
