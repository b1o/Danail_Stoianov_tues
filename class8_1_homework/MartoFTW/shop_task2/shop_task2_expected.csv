chicken,meskomers,9.30,09/11/2013
eggs,meskomers,2.50,01/11/2013
milk,meskomers,0.90,20/11/2013
cheese,meskomers,3.90,10/11/2013
boar meat,mesokomers,10.50,29/11/2013
coffe,nova brazilia,3.50,30/11/2013
peppers,vege inc,5.00,05/11/2013
